lib/semian/net_http.rb

```diff
@@ -0,0 +1,94 @@
+require 'semian'
+require 'semian/adapter'
+
+module Net
+  ProtocolError.include(::Semian::AdapterError)
+
+  class SemianError < ::Net::ProtocolError
+    def initialize(semian_identifier, *args)
+      super(*args)
+      @semian_identifier = semian_identifier
+    end
+  end
+
+  ResourceBusyError = Class.new(SemianError)
+  CircuitOpenError = Class.new(SemianError)
+end
+
+module Semian
+  module NetHTTP
+    include Semian::Adapter
+
+    ResourceBusyError = ::Net::ResourceBusyError
+    CircuitOpenError = ::Net::CircuitOpenError
+
+    def semian_identifier
+      @configuration ||= Semian::NetHTTP.retrieve_semian_configuration_by_host_port(address, port)
```

Comment:
Imagine if someone did something like this (which reminds me you should be able to set it like this, which should be possible by just setting it when `block_given?`, then you can remove the `#=` variant IMO):

```ruby
Semian::NetHTTP.raw_semian_options do |host, port|
  ["#{host}_#{port}_#{rand(2)}", some_default_options]
end
```

It's not really a legit use-case, but there could be other examples where the `semian_identifier` returned differ between multiple invocations. Currently we just don't support it at all, which is likely fine, I just want to make sure that's done on purpose. When I talk about switching closures, I mean overriding `raw_semian_options` at runtime, which might be later in the code where the above is:

```ruby
Semian::NetHTTP.raw_semian_options do |host, port|
  ["#{host}_#{port}_omghardcoded", some_default_options]
end
```

Now it wouldn't actually change the Semian identifier, since it's memoized on that object.